runtime/tools/python/ttrt/common/golden.py

```diff
@@ -148,11 +148,61 @@ def golden(binary, programContext, opContext):
             op_output_tensor, dtype=torch.float32
         ).flatten()
 
+        torch.save(golden_tensor_torch, f"{runtimeConfig.artifact_dir}/{loc}_golden.pt")
+        torch.save(output_tensor_torch, f"{runtimeConfig.artifact_dir}/{loc}_device.pt")
+
         _, _, cal_pcc, output_str = get_atol_rtol_pcc(
             golden_tensor_torch, output_tensor_torch
         )
 
         print(f"PCC={cal_pcc}")
         print(output_str)
+
+        results = {}
+        results["expected_pcc"] = runtimeConfig.pcc
+        results["actual_pcc"] = cal_pcc
+        results["atol"] = runtimeConfig.atol
+        results["rtol"] = runtimeConfig.rtol
+        results["allclose"] = torch.allclose(
+            golden_tensor_torch,
+            output_tensor_torch,
+            atol=runtimeConfig.atol,
+            rtol=runtimeConfig.rtol,
+        )
+        results["max"] = torch.max(
+            torch.abs(golden_tensor_torch - output_tensor_torch)
+        ).item()
+        results["mean_absolute_error"] = torch.mean(
+            torch.abs(golden_tensor_torch - output_tensor_torch)
+        ).item()
+        results["root_mean_square_error"] = torch.sqrt(
+            torch.mean((golden_tensor_torch - output_tensor_torch) ** 2)
+        ).item()
+        results["cosine_similarity"] = torch.nn.functional.cosine_similarity(
+            golden_tensor_torch.unsqueeze(0), output_tensor_torch.unsqueeze(0)
+        ).item()
+
+        # Create golden result json file if doesn't exist
+        golden_results_file_path = f"{runtimeConfig.artifact_dir}/golden_results.json"
+        if not os.path.exists(golden_results_file_path):
+            with open(golden_results_file_path, "w") as f:
+                json.dump({}, f)
+
+        with open(golden_results_file_path, "r") as f:
+            try:
+                data = json.load(f)
+                if not isinstance(data, dict):
+                    raise ValueError(
+                        f"Invalid JSON format: Expected a dict but got {type(data).__name__}"
+                    )
+            except json.JSONDecodeError:
+                data = {}
+
+        data[loc] = results
```

Comment:
This loc data is currently the regex form from the rest of the file, was @ctodTT not able to land his change that used the new `get_loc_info` property?

Reply:
waiting on it to land: https://github.com/tenstorrent/tt-mlir/pull/1488/files

It shouldn't change this implementation, as he still computes loc from proper location and this should be unique during flatbuffer generation.